docs/guides/transactions.md

```diff
@@ -1,6 +1,56 @@
 # Using transactions on the lakeFS file system
 
-A transaction defers file transfers and versioning operations to a queue, which is unwound sequentially on completion.
-Transactions are thread-safe and atomic, meaning that a single failure during any transaction function causes the entire transaction to abort.
+In addition to file operations, you can carry out versioning operations in your Python code using file system *transactions*.
 
-TODO: Add content
+A transaction is basically a context manager that collects all file uploads, defers them, and executes the uploads on completion of the transaction.
+They are an "all or nothing" proposition: If an error occurs during the transaction, none of the queued files are uploaded.
+
+The main features of the lakeFS file system transaction are:
+
+## Thread safety
+
+Through its use of `collections.deque` as a store for uploads, upload queueing and file transfers are thread-safe.
```

Comment:
A bit too much implementation detail for my liking, then again, there's not much more to say on the topic here. (I could even see leaving out the section altogether and just sticking "thread-safe" somewhere in the introduction)

Reply:
Maybe let's just move the section further towards the end of the page. While I agree it's useful information, it doesn't really help the reader right away (as the other sections do).